### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by koenbok**
A sweet little PR that should increase performance and save memory:

- Don't send events in the `Layer` constructor.
- Don't use explicit named setters/getters in `BaseClass`.

All tests pass.

### CODE REVIEW

File: framer/BaseClass.coffee [hunk 1]
```diff
@@ -23,19 +23,7 @@ class exports.BaseClass extends EventEmitter
...
 
-		# Set the getter/setter as setProperty on this object so we can access and override it easily
-		getName = "get#{capitalizeFirstLetter(propertyName)}"
-		@::[getName] = descriptor.get
-		descriptor.get = @::[getName]
-
-		if descriptor.set
-			setName = "set#{capitalizeFirstLetter(propertyName)}"
-			@::[setName] = descriptor.set
-			descriptor.set = @::[setName]
```
Comment (nvh) [inline framer/BaseClass.coffee]: Is this indeed never used? Nice!

File: framer/Layer.coffee [hunk 2]
```diff
@@ -81,6 +86,7 @@ class exports.Layer extends BaseClass
 		# Make sure we never call the constructor twice
 		throw Error("Layer.constructor #{@toInspect()} called twice") if @__constructed
 		@__constructed = true
+		@__constructing = true
```
Comment (nvh) [inline framer/Layer.coffee]: It's weird to set `@__constructing` after `@__constructed`, right? Maybe rename `@__constructed` to `@__constructorCalled` or something?

File: framer/BaseClass.coffee (hunk 1 above)
Comment (koenbok) [inline framer/BaseClass.coffee]: Nope.

File: framer/Layer.coffee (hunk 2 above)
Comment (koenbok) [inline framer/Layer.coffee]: 😘

**Comment by koenbok** [comment]
Not really measurable, but it saves around ~20 function calls per layer creation, so hey!

**Comment by nvh** [comment]
🎉

**Pull request merged**